packages/shared/src/contracts/agents/signing-coordinator.ts

```diff
@@ -77,6 +77,32 @@ export class SigningCoordinatorAgent {
     );
   }
 
+  public static async getCohortMultisigAddress(
+    provider: ethers.providers.Provider,
+    domain: Domain,
+    cohortId: number,
+    chainId: number,
+  ): Promise<string> {
+    const coordinator = await this.connectReadOnly(provider, domain);
+
+    // Get the SigningCoordinatorChild contract address for this chain
+    const childAddress = await coordinator.getSigningCoordinatorChild(chainId);
+
+    // Create a contract instance for the child (using generic Contract interface)
+    const childContract = new ethers.Contract(
+      childAddress,
+      [
+        // ABI for the cohortMultisigs function
+        'function cohortMultisigs(uint32) view returns (address)',
+      ],
+      provider,
```

Comment:
This only works if the child contract is on the same chain as the signing coordinator i.e. only the ETH chain. If the child chain is different the same provider cannot be used, because the provider is used for calling the signing coordinator, and therefore wont be on the same chain.

The best you can do here is return the address of the signing coordinator child contract and not actually make a call on the child contract if it is on a different chain, unless a separate provider is passed for the relevant child chain.